tests/components/generic_hygrostat/test_config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Test the generic hygrostat config flow."""
+
+from unittest.mock import patch
+
+from syrupy.assertion import SnapshotAssertion
+from syrupy.filters import props
+
+from homeassistant import config_entries
+from homeassistant.components.generic_hygrostat import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry, async_mock_service
+
+
+async def test_config_flow(hass: HomeAssistant, snapshot: SnapshotAssertion) -> None:
+    """Test the config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result == snapshot(name="init", exclude=props("data_schema"))
```

Comment:
Personal opinion, I don't like snapshots in config flows because they completely abstract away the thing that you want to test. The assertion becomes something like "the result should be something along these lines", but doesn't specify the critical points which do matter like `CREATE_ENTRY` or things like that

Reply:
True.. But I don't really care too much about what it returns. It's an internal feature of the of flow. I suppose i could maybe switch to props include, and just list one or two entries.

Reply:
I've reduced the snapshots so they are much easier to review and only include what we want to test.